code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -0,0 +1,258 @@
+// ***************************************
+// *********** Drain blood
+// ***************************************
+/datum/action/xeno_action/activable/drain
+	name = "Drain"
+	action_icon_state = "drain"
+	mechanics_text = "Stagger a marine and drain some of their blood. When used on a dead human, you heal gradually and don't gain blood."
+	cooldown_timer = 15 SECONDS
+	plasma_cost = 0
+	target_flags = XABB_MOB_TARGET
+
+/datum/action/xeno_action/activable/drain/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!ishuman(A) || issynth(A))
+		if(!silent)
+			to_chat(owner, span_xenowarning("We can't drain this!"))
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/carbon/human/H = A
+	if(!X.Adjacent(H))
+		if(!silent)
+			to_chat(owner, span_notice("We need to be next to our meal."))
+		return FALSE
+
+	if(H.stat == DEAD)
+		if(owner.do_actions)
+			return FALSE
+		if(!HAS_TRAIT(H, TRAIT_UNDEFIBBABLE))
+			if(!silent)
+				to_chat(owner, span_notice("We can only use fully dead hosts to heal."))
+			return FALSE
+		if(H.blood_volume < GORGER_REJUVENATE_BLOOD_DRAIN)
+			if(!silent)
+				to_chat(owner, span_notice("Our meal has no blood... How sad!"))
+			return FALSE
+		return TRUE
+
+	if(!.)
+		return
+
+	if(X.plasma_stored >= X.xeno_caste.plasma_max)
+		if(!silent)
+			to_chat(X, span_xenowarning("No need, we feel sated for now..."))
+		return FALSE
+
+/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	if(A.stat == DEAD)
+		var/mob/living/carbon/human/target = A
+		while(X.health < X.maxHealth && do_after(X, 2 SECONDS, TRUE, target, BUSY_ICON_HOSTILE))
```

Comment:
Since it's a hybrid health / sunder heal cycle, my suggestion would be allowing them drain a corpse to heal their sunder even if full on hp, since it's one of the two effects of this. Xenomorphs who for some reason don't want this behavior could always just cancel the drain once healed.
(was kinda weird to not have during the initial Gorger rounds)
```suggestion
		while((X.health < X.maxHealth || X.sunder) && do_after(X, 2 SECONDS, TRUE, target, BUSY_ICON_HOSTILE))
```

Reply:
Here the sunder heal is similar to resting, while the health heal is much more than resting. Due to this, it is preferable to not drain bodies for sunder heal since this uses up their blood, while you can be resting for free instead if you wanted sunder recovery.

Reply:
Prior to the most recent commits I'd have called it a valid tool for sunder recovery since it was still higher than base resting or retreating from non-weed areas to rest, though this is now effectively nulled by the number change. Feel free to just mark this resolved.